### PR TITLE
medical question pairs

### PR DESCRIPTION
**Pull request by debajyotidatta**
Medical question pairs dataset. Have used the knowledge that Curai's doctors labeled the data as well to generate some of the prompts.

### CODE REVIEW

File: templates/medical_questions_pairs/templates.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      |||
+
+
+      {{["No they do not mean the same thing.", "They mean the same thing."][label]}}'
+    name: basic_v2
+    reference: template about question pattern
+    task_template: true
+  7be2b267-8d5c-466b-9fd4-1fbbae442938: !Template
+    id: 7be2b267-8d5c-466b-9fd4-1fbbae442938
+    jinja: '{% if label == 0 or label == 1 %}
```
Comment (VictorSanh) [inline templates/medical_questions_pairs/templates.yaml:74]: i don't understand this condition. there can't be other values for label than 0 and 1 so this condition is not necessary

File: templates/medical_questions_pairs/templates.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+      {% if label == 0 %}
+
+      No
+
+      {% elif label == 1 %}
+
+      Yes
+
+      {% endif %}'
```
Comment (VictorSanh) [inline templates/medical_questions_pairs/templates.yaml:100]: Simpler: `{{["No", "Yes"][label]}}`

File: templates/medical_questions_pairs/templates.yaml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      \nQuestion 2: {{question_2}}\n\n{% endif %}\n\nAfter reading the two questions\
+      \ above:\n\n|||\n\n{% if label == 0 %} \nA clinician will not agree that they\
+      \ are paraphrases of each other.\n{% elif label == 1 %}\nA clinician will agree\
+      \ that they are paraphrases of each other.\n{% endif %}"
+    name: basic_v7
+    reference: Using knowledge of dataset creation
+    task_template: false
+  c8df74ce-0ae7-4e70-9322-aaf9921ae3b1: !Template
+    id: c8df74ce-0ae7-4e70-9322-aaf9921ae3b1
+    jinja: "The two questions are either similar dissimilar questions in the medical\
```
Comment (VictorSanh) [inline templates/medical_questions_pairs/templates.yaml:124]: `{{"similar""}} or {{"dissimilar"}}`

File: templates/medical_questions_pairs/templates.yaml (hunk 1 above)
Comment (VictorSanh) [reply templates/medical_questions_pairs/templates.yaml:74]: found 5 occurences of this. same comment

File: templates/medical_questions_pairs/templates.yaml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    reference: context of healthcare
+    task_template: true
+  854ebbe0-8968-4967-a346-4e4d6f98cf73: !Template
+    id: 854ebbe0-8968-4967-a346-4e4d6f98cf73
+    jinja: "{% if label == 0 or label == 1 %}\n\nQuestion 1: {{question_1}}\n\n{%\
+      \ endif %}\n\nOne possible {{[\"incorrect\", \"correct\"][label]}} way of paraphrasing\
+      \ the same question is : \n\n|||\n\nQuestion 2: {{question_2}}"
+    name: basic_v4
+    reference: ''
+    task_template: false
```
Comment (VictorSanh) [inline templates/medical_questions_pairs/templates.yaml:111]: let's do something like:

```jinja
{% if label == 1 %}
Question 1: {{question_1}}
One possible way of paraphrasing the same question is:|||
{{question 2}}
{% endif %}
```

Everything after the `|||` should be the label/genration, if you formulate a label as the 2nd part of a sentence, have the 1st part of the sentence inside the prompt

File: templates/medical_questions_pairs/templates.yaml (hunk 2 above)
Comment (VictorSanh) [reply templates/medical_questions_pairs/templates.yaml:100]: you can replace a couple of things using this compact format (basic_v5 for instance)

File: templates/medical_questions_pairs/templates.yaml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  b388913a-9b0d-43a1-8bf9-83319ebf38b2: !Template
+    id: b388913a-9b0d-43a1-8bf9-83319ebf38b2
+    jinja: "{% if label == 0 or label == 1 %}\n\nQuestion 1: {{question_1}}\n\nand\n\
+      \nQuestion 2: {{question_2}}\n\n{% endif %}\n\nAfter reading the two questions\
+      \ above:\n\n|||\n\n{% if label == 0 %} \nA clinician will not agree that they\
+      \ are paraphrases of each other.\n{% elif label == 1 %}\nA clinician will agree\
```
Comment (VictorSanh) [inline templates/medical_questions_pairs/templates.yaml:117]: move `A clinician will ` inside the prompt

**Review by VictorSanh — CHANGES_REQUESTED** [review]
Thank you @debajyotidatta ! please adress my comments

`glue/qqp`'s templates are a good example of creativity. can you adapt the most funky ones? this is very similar task

File: templates/medical_questions_pairs/templates.yaml [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+      \ that they are paraphrases of each other.\n{% endif %}"
+    name: basic_v7
+    reference: Using knowledge of dataset creation
+    task_template: false
```
Comment (VictorSanh) [inline templates/medical_questions_pairs/templates.yaml:121]: should be True

File: templates/medical_questions_pairs/templates.yaml [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  316f2ff7-45f8-4997-9c5f-dfe5fb7f9808: !Template
+    id: 316f2ff7-45f8-4997-9c5f-dfe5fb7f9808
+    jinja: "{% if label == 0 or label == 1 %}\n\nQuestion 1: {{question_1}}\n\nand\n\
+      \nQuestion 2: {{question_2}}\n\n{% endif %}\n\nIs it True or False that the\
+      \ two questions above are paraphrases of each other?\n\n|||\n\n {{[\"False\"\
+      , \"True\"][label]}} "
+    name: basic_v6
+    reference: ''
+    task_template: false
```
Comment (VictorSanh) [inline templates/medical_questions_pairs/templates.yaml:52]: Should be true

File: templates/medical_questions_pairs/templates.yaml [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+      {% elif label == 1 %}
+
+
+      This question is a paraphrase of the previous question.
+
+
+      {% endif %}'
+    name: basic_v5
+    reference: ''
+    task_template: false
```
Comment (VictorSanh) [inline templates/medical_questions_pairs/templates.yaml:43]: should be true

**Comment by debajyotidatta** [comment]
@VictorSanh : Addressed the comments. Thanks for the reference to glue/qqp. I adapted most of them and added a new template.

**Review by VictorSanh — APPROVED** [review]
thank you! merging now, please update the spreadsheet!

**Pull request merged**